Лабораторная работа 1/task1.py

```diff
@@ -0,0 +1,130 @@
+import doctest
+
+
+class Groups:
+    def __init__(self, order: int, abelian_check: bool = False):
+        """
+        Создание группы
+        :param order: Порядок данной группы
+        :param abelian_check: Выбор, будет ли группа абелевой
+        """
+        if not isinstance(order, int):
+            raise TypeError('Порядок группы - натуральное число')
+        if not order >= 0:
+            raise ValueError('Порядок группы - натуральное число')
+        self.order = order
+
+        if not isinstance(abelian_check, bool):
+            raise TypeError
+        self.abelian_check = abelian_check
+
+    def quotient(self, normal_sub):
+        """
+        Поиск факторгруппы по заданной нормальной подгруппе
+        :param normal_sub: Нормальная подгруппа
+        :raise ValueError: Порядок группы должен делиться на порядок подгруппы, иначе - вызываем ошибку
+        :return: Полученная факторгруппа
+        Пример:
+        >>> G = Groups(6)
+        >>> H = Groups(2)
+        >>> G_H = G.quotient(H)
+        """
+        if not isinstance(normal_sub, Groups):
+            # когда пишу аннотацию типа normal_sub: Groups, выдает ошибку, будто класс Groups ещё не создан
+            # а когда пишу проверку принадлежности аргумента к классу Groups через isinstance - никакой ошибки нет
+            # почему так?
```

Comment:
https://stackoverflow.com/questions/40049016/using-the-class-as-a-type-hint-for-arguments-in-its-methods